File: benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/ThroughputOperation.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.Benchmark.Mode;
import de.hhu.bsinfo.observatory.benchmark.result.ThroughputMeasurement;

abstract class ThroughputOperation extends Operation {

    ThroughputOperation(Benchmark benchmark, Mode mode, int operationCount, int operationSize) {
        super(benchmark, mode, new ThroughputMeasurement(operationCount, operationSize));
    }

    @Override
    protected ThroughputMeasurement getMeasurement() {
        return (ThroughputMeasurement) super.getMeasurement();
    }
}

File: cpp/src/observatory/operation/ThroughputOperation.cpp
#include "ThroughputOperation.h"

namespace Observatory {

ThroughputOperation::ThroughputOperation(Benchmark *benchmark, Benchmark::Mode mode, uint32_t operationCount, uint32_t operationSize) :
        Operation(benchmark, mode, std::make_shared<ThroughputMeasurement>(operationCount, operationSize)) {

}

ThroughputMeasurement& ThroughputOperation::getMeasurement() const {
    return dynamic_cast<ThroughputMeasurement&>(Operation::getMeasurement());
}

const char *ThroughputOperation::getClassName() const {
    return "ThroughputOperation";
}

}